lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -1193,11 +1197,17 @@ def install_puppet_agent_dev_repo_on( hosts, opts )
               raise "No repository installation step for #{variant} yet..."
             end
 
-            onhost_copied_file = File.join(onhost_copy_base, release_file)
-            fetch_http_file( release_path, release_file, copy_dir_local)
-            scp_to host, File.join(copy_dir_local, release_file), onhost_copy_base
+            if host['platform'] =~ /eos/
```

Comment:
Again, not a criticism of this PR, which is limited in scope, but I'm not excited about how liberally some of our code (`foss_utils` being an example) devolves into strings of `if` blocks. This makes me think again that we should be looking at a different breakdown of domain concepts that might enable us to use inheritance or delegation instead of conditionals.